code/modules/projectiles/guns/ballistic/revolver.dm

```diff
@@ -154,6 +154,121 @@
 
 	mag_type = /obj/item/ammo_box/magazine/internal/cylinder/rev762
 
+#define HAMMER_CLOSED 0
+#define HAMMER_RELOAD 1
+#define HAMMER_FULL 2
+
+/obj/item/gun/ballistic/revolver/single_action
+	name = ".44 Magnum single action revolver"
+	desc = "A single action fixed cylinder revolver. Ineffective in the hands of an inexperienced user due to the lack of automation found in more recently designed weapons, though its robust construction allows it to chamber powerful ammunition. Uses .44 Magnum ammo."
+	icon_state = "single-action"
+	mag_type = /obj/item/ammo_box/magazine/internal/cylinder/rev44
+	///current hampter position
+	var/hammer_set = HAMMER_CLOSED
+
+/obj/item/gun/ballistic/revolver/single_action/update_icon()
+	cut_overlays()
+	switch(hammer_set)
+		if(HAMMER_CLOSED)
+			add_overlay("[icon_state]_closed")
+		if(HAMMER_RELOAD)
+			add_overlay("[icon_state]_half")
+		if(HAMMER_FULL)
+			add_overlay("[icon_state]_full")
+
+
+/obj/item/gun/ballistic/revolver/single_action/examine(mob/user)
+	. = ..()
+	. += "Use in hand to switch between reloading and firing. Currently [hammer_set == HAMMER_RELOAD ? "reloading" : "firing"]."
+	if(hammer_set == HAMMER_RELOAD)
+		var/obj/item/ammo_casing/casing = magazine.get_round(TRUE, FALSE)
+		. += "The currently accessible chamber is [casing ? casing.BB ? "loaded" : "spent" : "empty"]."
+
+	. += "While firing, attempting to shoot will first pull the hammer back. If the hammer is pulled back, shooting will shoot."
+	. += "While reloading, attempting to shoot will rotate the cylinder and eject spent casings, Alt-Click to eject any bullet in the accessible cylinder, spent or not."
+
+/obj/item/gun/ballistic/revolver/single_action/afterattack(atom/target, mob/living/user, flag, params)
+	if(hammer_set == HAMMER_RELOAD)
+		var/obj/item/ammo_casing/casing = magazine.get_round(TRUE, FALSE)
+		var/obj/item/ammo_box/magazine/internal/cylinder/cylinder = magazine
+		user.changeNext_move(CLICK_CD_RANGE)
+		if(casing && !casing.BB)
+			casing.forceMove(drop_location())
+			magazine.get_round(FALSE, FALSE) //this removes the round from our cylinder
+			to_chat(user, span_warning("You push [casing] out of [src]."))
+			return
+		to_chat(user, span_warning("You rotate [src]'s cylinder."))
+		cylinder.rotate()
+		casing = magazine.get_round(TRUE, FALSE)
+		balloon_alert(user, casing ? casing.BB ? "Chamber loaded" : "Chamber spent" : "Chamber empty")
```

Comment:
balloon alerts normally have the first letter not capitalized, unless it's something specific, just want to remain consistent